bin/check-megaraid-sas-status.rb

```diff
@@ -53,12 +53,18 @@ class CheckMegaRAID < Sensu::Plugin::Check::CLI
   def run
     have_error = false
     error = ''
-    # get number of virtual drives
-    `#{config[:megaraidcmd]} -LDGetNum -a#{config[:controller]} `
-    (0..$CHILD_STATUS.exitstatus - 1).each do |i|
+    # get list of virtual drives
+    stdout = `#{config[:megaraidcmd]} -LDInfo -LALL -a#{config[:controller]} `.split("\n")
+    virtual_drives = []
+    stdout.each do |line|
+      if line =~ /^Virtual Drive:/
+        virtual_drives << line.gsub(/^Virtual Drive: (\d+) .*$/, '\1')
+      end
+    end
+    virtual_drives.each do |i|
       # and check them in turn
       stdout = `#{config[:megaraidcmd]} -LDInfo -L#{i} -a#{config[:controller]} `
-      unless Regexp.new('State\s*:\s*Optimal').match?(stdout)
+      unless Regexp.new(/State\s+:\s+Optimal/).match(stdout)
```

Comment:
the `.match?` while is indeed faster and more resource conscious as it discards the actual matchdata and only returns a boolean value it did not exist prior to ruby `2.4`. I will add a comment to change that back when we drop ruby `< 2.4` support (which we won't do until after ruby 2.4 is EOL).